### DISCUSSION TITLE
Reduction of coordination number

### DISCUSSION THREAD

**Discussion by jayaraj777** [PMEAL/OpenPNM#2715] [Q&A] [answered]
Hi,
I am trying to create a cubic network of size [20,20,51]. I found the average coordination number to be 5.7 but I want it to reduce to 3.9. I also want to eliminate dead pores and clusters in the network. I have used the command op.topotools.reduce_coordination(network=pn, z=3.9) for reducing the coordination number but I am not sure whether it got reduced. Also, I am not able to use the trim command for eliminating dead-end pores and clusters. I am new to OpenPNM. Please suggest me a solution to these problems.  

```python
import openpnm as op
from openpnm.utils import HealthDict
import matplotlib.pyplot as plt
import numpy as np
import scipy. stats as spst
op.visualization.set_mpl_style()
pn = op.network.Cubic(shape=[20,20,51], spacing=1)
fig = op.visualization.plot_connections(network=pn)
geo = op.models.collections.geometry.spheres_and_cylinders

pn.add_model_collection(geo)
pn.regenerate_models()
print(pn)

neighbor_pores= pn.find_neighbor_pores(pn.pores())
num_neighbors= pn.num_neighbors(pn.pores(), flatten=False)
print (num_neighbors)

Avg_Cnum= np.mean(num_neighbors)
print(Avg_Cnum)

op.topotools.reduce_coordination(network=pn, z=3.9)     # Reducing average coordination number to 3.9

Ps=pn.num_neighbors(pores=pn.Ps)
op.topotools.trim(pn,pores=Ps==1)
while (Ps==1).any():
    Ps=pn.num_neighbors(pores=pn.Ps)
    op.topotools.trim(pn,pores=Ps==1)
print("Number of pores after elimination=", pn.num_pores())

h=op.utils.check_network_health(network=pn)
print(h)
```

**Comment by jayaraj777** [comment]
Kindly consider this a gentle reminder.


Thanks
Jayaraj

**Comment by jgostick** [comment, marked as answer]
The `reduce_coordination` function creates a new label on the network (I think it's `'pore.to_trim'`).  You then do:
``` python
Ps = pn.pores('to_trim')
op.topotools.trim(network=pn, pores=Ps)
```

Your network will now have the desired coordination number.  

Note that the `reduce_coordination` function is a bit of a hack...it computes a minimum spanning tree using random weights, then randomly selects throats that are *not* part of the spanning tree for deletion.  This will give you the correct mean coordination, but has not control at all over the distribution of the coordination. The point of using the spanning tree is that the network will remain connected after the deletion.

**Reply by jayaraj777** [reply]
Thanks, I was able to solve my problem.


Jayaraj
